### PR TITLE
Implement Github CI (Actions)

### PR DESCRIPTION
**Pull request by Mikescops**
Implement Github CI so it can replace Travis for faster test checks and better integration with Github.

### CODE REVIEW

**Comment by Mikescops** [comment]
Shall we remove Travis too ?

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Node CI
+
+on: [push]
```
Comment (highfivedenis) [inline .github/workflows/nodejs.yml:3]: As a general comment I'm not highly familiar with Github Actions yet but:
- Why only on push and not on `pull_request` too?

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [13.x]
+
+    steps:
+    - uses: actions/checkout@v1
```
Comment (highfivedenis) [inline .github/workflows/nodejs.yml:15]: It seems that there have release a `checkout@v2` already

File: .github/workflows/nodejs.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: install, build, and test
+      run: |
```
Comment (highfivedenis) [inline .github/workflows/nodejs.yml:21]: Naturally, I would prefer to split the 2 commands in two separate runs:
```
 - run: yarn install
 - run: npm run ci
```

**Review by highfivedenis — CHANGES_REQUESTED** [review]
Thanks for doing this! Could you remove the Travis config file in this PR too?

File: .github/workflows/nodejs.yml (hunk 1 above)
Comment (Mikescops) [reply .github/workflows/nodejs.yml:3]: I don't know, i guess it's the same

Comment (ariasuni) [reply .github/workflows/nodejs.yml:3]: If you already have `push`, `pull_request` is only useful if you create the branch on a different repository. It doesn’t cost anything to add `pull_request` if we expect someone someday to contribute to our repository

https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-events-for-forked-repositories-1

> When you create a pull request from a forked repository to the base repository, GitHub sends the `pull_request` event to the base repository and no pull request events occur on the forked repository.
> 
> Workflows don't run on forked repositories by default. You must enable GitHub Actions in the **Actions** tab of the forked repository.

**Comment by ariasuni** [comment]
I’m fiddling with GitHub Actions these days so I’d be happy to resurrect this PR so that it can finally be merged if that is still what you want.

**Comment by Mikescops** [comment]
@ariasuni sure you can

**Comment by ariasuni** [comment]
- Dropped Node 13 since [it’s unsupported](https://nodejs.org/en/about/releases/)
- Test Node 14 et 16 (two current LTS versions)
- CI only started if we touch code, tests or build/lint configuration
- Use `actions/checkout@v2` and `actions/setup-node@v2` instead of their v1
- Removed the Travis CI configuration file

It seems like some configuration you be changed for Travis CI to be fully disabled.

**Comment by ariasuni** [comment]
By the way I’m in vacations tonight and will come back the 18th.

**Review by highfivedenis — APPROVED** [review]

**Comment by highfivedenis** [comment]
Thank you @ariasuni and @Mikescops!
Unfortunately due to the age of the PR it still has the Travis CI check in it. Could you recreate the PR?
Thanks!

**Comment by ariasuni** [comment]
> Unfortunately due to the age of the PR it still has the Travis CI check in it. Could you recreate the PR?

Yeah, that’s not it: see #36, it still needs Travis CI to pass. Someone probably needs to disable Travis on GitHub or Travis’ side for it to work.

**Pull request merged**